Цель работы: реализовать базовый анализ датасета пассажиров Титаника, вывести такие параметры датасета как размерность, наименование колонок, уникальные значения, реализовать сортировку, удаление, замену, удаление дубликатов, провести анализ с помощью info, describe, провести выборку данных с помощью loc.

In [55]:
import pandas as pd

titanic = pd.read_csv("titanic.csv")

titanic

pclass  survived                                             name  \
0          1         1                    Allen, Miss. Elisabeth Walton   
1          1         1                   Allison, Master. Hudson Trevor   
2          1         0                     Allison, Miss. Helen Loraine   
3          1         0             Allison, Mr. Hudson Joshua Creighton   
4          1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1304       3         0                             Zabour, Miss. Hileni   
1305       3         0                            Zabour, Miss. Thamine   
1306       3         0                        Zakarian, Mr. Mapriededer   
1307       3         0                              Zakarian, Mr. Ortin   
1308       3         0                               Zimmerman, Mr. Leo   

         sex      age  sibsp  parch  ticket      fare    cabin embarked boat  \
0     female  29.0000      0      0   24160  211.3375       B5        S    2   
1       male   0.9167      1      2  113781  151.5500  C22 C26        S   11   
2     female   2.0000      1      2  113781  151.5500  C22 C26        S  NaN   
3       male  30.0000      1      2  113781  151.5500  C22 C26        S  NaN   
4     female  25.0000      1      2  113781  151.5500  C22 C26        S  NaN   
...      ...      ...    ...    ...     ...       ...      ...      ...  ...   
1304  female  14.5000      1      0    2665   14.4542      NaN        C  NaN   
1305  female      NaN      1      0    2665   14.4542      NaN        C  NaN   
1306    male  26.5000      0      0    2656    7.2250      NaN        C  NaN   
1307    male  27.0000      0      0    2670    7.2250      NaN        C  NaN   
1308    male  29.0000      0      0  315082    7.8750      NaN        S  NaN   

       body                        home.dest  
0       NaN                     St Louis, MO  
1       NaN  Montreal, PQ / Chesterville, ON  
2       NaN  Montreal, PQ / Chesterville, ON  
3     135.0  Montreal, PQ / Chesterville, ON  
4       NaN  Montreal, PQ / Chesterville, ON  
...     ...                              ...  
1304  328.0                              NaN  
1305    NaN                              NaN  
1306  304.0                              NaN  
1307    NaN                              NaN  
1308    NaN                              NaN  

[1309 rows x 14 columns]

1. Вывод данных.
Выведем полный датасет со всеми данными. Всего 1309 пассажиров, это, к слову, меньше, чем было на титанике, но мы сделаем вид, что так и надо. Более подробный разбор датасета будет в пункте про наименование колонок.

In [34]:
titanic.shape

(1309, 14)

2. Размерность массива
Размерность показывает количество строк и столбцов таблицы. В нашем случае общее количество ячеек будет 18326, столько понадобится индексов, чтобы определить каждый элемент массива. 

In [35]:
titanic.columns

Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')

3. Наименования колонок
В нашем датасете указаны такие данные пассажиров, как имя, пол, возраст, класс билета, цена билета, количество сестёр/братьев или супруг на борту, количество родителей/детей на борту, номер каюты, порт посадки, номер лодки, номер тела, пункт отправления и назначения, и главный показатель - выжил пассажир или нет. В дальнейшем мы будем все остальные показатели сравнивать с тем, выжил пассажир или нет.

In [36]:
titanic['pclass'].unique()

array([1, 2, 3], dtype=int64)

In [37]:
titanic['survived'].unique()

array([1, 0], dtype=int64)

In [11]:
titanic['name'].unique()

array(['Allen, Miss. Elisabeth Walton', 'Allison, Master. Hudson Trevor',
       'Allison, Miss. Helen Loraine', ..., 'Zakarian, Mr. Mapriededer',
       'Zakarian, Mr. Ortin', 'Zimmerman, Mr. Leo'], dtype=object)

In [12]:
titanic['sex'].unique()

array(['female', 'male'], dtype=object)

In [13]:
titanic['age'].unique()

array([29.    ,  0.9167,  2.    , 30.    , 25.    , 48.    , 63.    ,
       39.    , 53.    , 71.    , 47.    , 18.    , 24.    , 26.    ,
       80.    ,     nan, 50.    , 32.    , 36.    , 37.    , 42.    ,
       19.    , 35.    , 28.    , 45.    , 40.    , 58.    , 22.    ,
       41.    , 44.    , 59.    , 60.    , 33.    , 17.    , 11.    ,
       14.    , 49.    , 76.    , 46.    , 27.    , 64.    , 55.    ,
       70.    , 38.    , 51.    , 31.    ,  4.    , 54.    , 23.    ,
       43.    , 52.    , 16.    , 32.5   , 21.    , 15.    , 65.    ,
       28.5   , 45.5   , 56.    , 13.    , 61.    , 34.    ,  6.    ,
       57.    , 62.    , 67.    ,  1.    , 12.    , 20.    ,  0.8333,
        8.    ,  0.6667,  7.    ,  3.    , 36.5   , 18.5   ,  5.    ,
       66.    ,  9.    ,  0.75  , 70.5   , 22.5   ,  0.3333,  0.1667,
       40.5   , 10.    , 23.5   , 34.5   , 20.5   , 30.5   , 55.5   ,
       38.5   , 14.5   , 24.5   , 60.5   , 74.    ,  0.4167, 11.5   ,
       26.5   ])

In [14]:
titanic['sibsp'].unique()

array([0, 1, 2, 3, 4, 5, 8], dtype=int64)

In [15]:
titanic['parch'].unique()

array([0, 2, 1, 4, 3, 5, 6, 9], dtype=int64)

In [16]:
titanic['ticket'].unique()

array(['24160', '113781', '19952', '13502', '112050', '11769', 'PC 17609',
       'PC 17757', 'PC 17477', '19877', '27042', 'PC 17318', 'PC 17558',
       '11813', '13050', '11751', '111369', 'PC 17483', '13905', '11967',
       'PC 17760', '110564', '113784', '112277', '36928', '113783',
       '110489', 'PC 17608', '113505', '111427', '113054', 'PC 17591',
       '112379', 'PC 17610', '16966', '113050', '113798', 'PC 17476',
       'PC 17606', 'PC 17755', '695', '113059', '113760', '19924',
       '17770', 'W.E.P. 5734', '113806', '110152', 'PC 17594', '112051',
       '13508', '110465', '5727', 'PC 17756', '11770', '113791',
       'WE/P 5735', '112901', 'PC 17599', '113055', '113804',
       'F.C. 12750', '17474', '33638', 'PC 17761', '11755', 'PC 17485',
       'PC 17580', '11767', '36947', 'PC 17531', 'PC 17598', '17421',
       'PC 17474', '113051', '19950', '113778', 'PC 17611', '17765',
       '13568', '13567', '112058', '113803', '111320', '113503', '112378',
       'PC 17593

In [17]:
titanic['fare'].unique()

array([211.3375, 151.55  ,  26.55  ,  77.9583,   0.    ,  51.4792,
        49.5042, 227.525 ,  69.3   ,  78.85  ,  30.    ,  25.925 ,
       247.5208,  76.2917,  75.2417,  52.5542, 221.7792,  26.    ,
        91.0792, 135.6333,  35.5   ,  31.    , 164.8667, 262.375 ,
        55.    ,  30.5   ,  50.4958,  39.6   ,  27.7208, 134.5   ,
        26.2875,  27.4458, 512.3292,   5.    ,  47.1   , 120.    ,
        61.175 ,  53.1   ,  86.5   ,  29.7   , 136.7792,  52.    ,
        25.5875,  83.1583,  25.7   ,  71.    ,  71.2833,  57.    ,
        81.8583, 106.425 ,  56.9292,  78.2667,  31.6792,  31.6833,
       110.8833,  26.3875,  27.75  , 263.    , 133.65  ,  49.5   ,
        79.2   ,  38.5   , 211.5   ,  59.4   ,  89.1042,  34.6542,
        28.5   , 153.4625,  63.3583,  55.4417,  76.7292,  42.4   ,
        83.475 ,  93.5   ,  42.5   ,  51.8625,  50.    ,  57.9792,
        90.    ,  30.6958,  80.    ,  28.7125,  25.9292,  39.4   ,
        45.5   , 146.5208,  82.1708,  57.75  , 113.275 ,  26.2

In [18]:
titanic['cabin'].unique()

array(['B5', 'C22 C26', 'E12', 'D7', 'A36', 'C101', nan, 'C62 C64', 'B35',
       'A23', 'B58 B60', 'D15', 'C6', 'D35', 'C148', 'C97', 'B49', 'C99',
       'C52', 'T', 'A31', 'C7', 'C103', 'D22', 'E33', 'A21', 'B10', 'B4',
       'E40', 'B38', 'E24', 'B51 B53 B55', 'B96 B98', 'C46', 'E31', 'E8',
       'B61', 'B77', 'A9', 'C89', 'A14', 'E58', 'E49', 'E52', 'E45',
       'B22', 'B26', 'C85', 'E17', 'B71', 'B20', 'A34', 'C86', 'A16',
       'A20', 'A18', 'C54', 'C45', 'D20', 'A29', 'C95', 'E25', 'C111',
       'C23 C25 C27', 'E36', 'D34', 'D40', 'B39', 'B41', 'B102', 'C123',
       'E63', 'C130', 'B86', 'C92', 'A5', 'C51', 'B42', 'C91', 'C125',
       'D10 D12', 'B82 B84', 'E50', 'D33', 'C83', 'B94', 'D49', 'D45',
       'B69', 'B11', 'E46', 'C39', 'B18', 'D11', 'C93', 'B28', 'C49',
       'B52 B54 B56', 'E60', 'C132', 'B37', 'D21', 'D19', 'C124', 'D17',
       'B101', 'D28', 'D6', 'D9', 'B80', 'C106', 'B79', 'C47', 'D30',
       'C90', 'E38', 'C78', 'C30', 'C118', 'D36', 'D48', 'D47', '

In [19]:
titanic['embarked'].unique()

array(['S', 'C', nan, 'Q'], dtype=object)

In [20]:
titanic['boat'].unique()

array(['2', '11', nan, '3', '10', 'D', '4', '9', '6', 'B', '8', 'A', '5',
       '7', 'C', '14', '5 9', '13', '1', '15', '5 7', '8 10', '12', '16',
       '13 15 B', 'C D', '15 16', '13 15'], dtype=object)

In [21]:
titanic['body'].unique()

array([ nan, 135.,  22., 124., 148., 208., 172., 269.,  62., 133., 275.,
       147., 110., 307.,  38.,  80.,  45., 258., 126., 292., 175., 249.,
       230., 122., 263., 234., 189., 166., 207., 232.,  16., 109.,  96.,
        46., 245., 169., 174.,  97.,  18., 130.,  17., 295., 286., 236.,
       322., 297., 155., 305.,  19.,  75.,  35., 256., 149., 283., 165.,
       108., 121.,  52., 209., 271.,  43.,  15., 101., 287.,  81., 294.,
       293., 190.,  72., 103.,  79., 259., 260., 142., 299., 171.,   9.,
       197.,  51., 187.,  68.,  47.,  98., 188.,  69., 306., 120., 143.,
       156., 285.,  37.,  58.,  70., 196., 153.,  61.,  53., 201., 309.,
       181., 173.,  89.,   4., 206., 327., 119.,   7.,  32.,  67., 284.,
       261., 176.,  50.,   1., 255., 298., 314.,  14., 131., 312., 328.,
       304.])

In [22]:
titanic['home.dest'].unique()

array(['St Louis, MO', 'Montreal, PQ / Chesterville, ON', 'New York, NY',
       'Hudson, NY', 'Belfast, NI', 'Bayside, Queens, NY',
       'Montevideo, Uruguay', 'Paris, France', nan, 'Hessle, Yorks',
       'Montreal, PQ', 'Winnipeg, MN', 'San Francisco, CA',
       'Dowagiac, MI', 'Stockholm, Sweden / Washington, DC',
       'Trenton, NJ', 'Glen Ridge, NJ', 'Youngstown, OH',
       'Birkdale, England Cleveland, Ohio', 'London / Winnipeg, MB',
       'Cooperstown, NY', 'St Leonards-on-Sea, England Ohio',
       'Los Angeles, CA', 'Pomeroy, WA', 'Omaha, NE', 'Philadelphia, PA',
       'Denver, CO', 'Belmont, MA', 'Washington, DC',
       'Austria-Hungary / Germantown, Philadelphia, PA',
       'Germantown, Philadelphia, PA', 'Bryn Mawr, PA',
       'Ascot, Berkshire / Rochester, NY', 'Little Onn Hall, Staffs',
       'Amenia, ND', 'New York, NY / Ithaca, NY', 'London, England',
       'Liverpool, England / Belfast', 'Stoughton, MA', 'Victoria, BC',
       'Lakewood, NJ', 'Roachdale, I

In [38]:
titanic.nunique()

pclass          3
survived        2
name         1307
sex             2
age            98
sibsp           7
parch           8
ticket        929
fare          281
cabin         186
embarked        3
boat           27
body          121
home.dest     369
dtype: int64

4. Уникальные значения
Выводим уникальные значения для каждой колонки, а также общее количество уникальных значений в датасете.

In [39]:
titanic.sort_values(by='fare')

pclass  survived                                               name  \
363        2         0                              Campbell, Mr. William   
223        1         0                      Parr, Mr. William Henry Marsh   
170        1         1                            Ismay, Mr. Joseph Bruce   
1254       3         1                       Tornquist, Mr. William Henry   
896        3         0                                Johnson, Mr. Alfred   
...      ...       ...                                                ...   
302        1         1                                   Ward, Miss. Anna   
50         1         1  Cardeza, Mrs. James Warburton Martinez (Charlo...   
49         1         1                 Cardeza, Mr. Thomas Drake Martinez   
183        1         1                             Lesurer, Mr. Gustave J   
1225       3         0                                 Storey, Mr. Thomas   

         sex   age  sibsp  parch    ticket      fare        cabin embarked  \
363     male   NaN      0      0    239853    0.0000          NaN        S   
223     male   NaN      0      0    112052    0.0000          NaN        S   
170     male  49.0      0      0    112058    0.0000  B52 B54 B56        S   
1254    male  25.0      0      0      LINE    0.0000          NaN        S   
896     male  49.0      0      0      LINE    0.0000          NaN        S   
...      ...   ...    ...    ...       ...       ...          ...      ...   
302   female  35.0      0      0  PC 17755  512.3292          NaN        C   
50    female  58.0      0      1  PC 17755  512.3292  B51 B53 B55        C   
49      male  36.0      0      1  PC 17755  512.3292  B51 B53 B55        C   
183     male  35.0      0      0  PC 17755  512.3292         B101        C   
1225    male  60.5      0      0      3701       NaN          NaN        S   

     boat   body                                       home.dest  
363   NaN    NaN                                         Belfast  
223   NaN    NaN                                         Belfast  
170     C    NaN                                       Liverpool  
1254   15    NaN                                             NaN  
896   NaN    NaN                                             NaN  
...   ...    ...                                             ...  
302     3    NaN                                             NaN  
50      3    NaN                    Germantown, Philadelphia, PA  
49      3    NaN  Austria-Hungary / Germantown, Philadelphia, PA  
183     3    NaN                                             NaN  
1225  NaN  261.0                                             NaN  

[1309 rows x 14 columns]

In [40]:
titanic.sort_values(by='age')

pclass  survived                                     name     sex  \
763        3         1  Dean, Miss. Elizabeth Gladys "Millvina"  female   
747        3         0  Danbom, Master. Gilbert Sigvard Emanuel    male   
1240       3         1          Thomas, Master. Assad Alexander    male   
427        2         1                Hamalainen, Master. Viljo    male   
1111       3         0           Peacock, Master. Alfred Edward    male   
...      ...       ...                                      ...     ...   
1293       3         0        Williams, Mr. Howard Hugh "Harry"    male   
1297       3         0                   Wiseman, Mr. Phillippe    male   
1302       3         0                        Yousif, Mr. Wazli    male   
1303       3         0                    Yousseff, Mr. Gerious    male   
1305       3         0                    Zabour, Miss. Thamine  female   

         age  sibsp  parch              ticket     fare cabin embarked boat  \
763   0.1667      1      2           C.A. 2315  20.5750   NaN        S   10   
747   0.3333      0      2              347080  14.4000   NaN        S  NaN   
1240  0.4167      0      1                2625   8.5167   NaN        C   16   
427   0.6667      1      1              250649  14.5000   NaN        S    4   
1111  0.7500      1      1  SOTON/O.Q. 3101315  13.7750   NaN        S  NaN   
...      ...    ...    ...                 ...      ...   ...      ...  ...   
1293     NaN      0      0            A/5 2466   8.0500   NaN        S  NaN   
1297     NaN      0      0          A/4. 34244   7.2500   NaN        S  NaN   
1302     NaN      0      0                2647   7.2250   NaN        C  NaN   
1303     NaN      0      0                2627  14.4583   NaN        C  NaN   
1305     NaN      1      0                2665  14.4542   NaN        C  NaN   

      body                   home.dest  
763    NaN  Devon, England Wichita, KS  
747    NaN                 Stanton, IA  
1240   NaN                         NaN  
427    NaN                 Detroit, MI  
1111   NaN                         NaN  
...    ...                         ...  
1293   NaN                         NaN  
1297   NaN                         NaN  
1302   NaN                         NaN  
1303   NaN                         NaN  
1305   NaN                         NaN  

[1309 rows x 14 columns]

5. Сортировка.
Отсортируем наш датасет по стоимости билета и по возрасту.

In [41]:
titanic.drop(titanic.columns[[5, 6, 7 , 9, 10, 11, 12, 13]], axis=1)

pclass  survived                                             name  \
0          1         1                    Allen, Miss. Elisabeth Walton   
1          1         1                   Allison, Master. Hudson Trevor   
2          1         0                     Allison, Miss. Helen Loraine   
3          1         0             Allison, Mr. Hudson Joshua Creighton   
4          1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1304       3         0                             Zabour, Miss. Hileni   
1305       3         0                            Zabour, Miss. Thamine   
1306       3         0                        Zakarian, Mr. Mapriededer   
1307       3         0                              Zakarian, Mr. Ortin   
1308       3         0                               Zimmerman, Mr. Leo   

         sex      age      fare  
0     female  29.0000  211.3375  
1       male   0.9167  151.5500  
2     female   2.0000  151.5500  
3       male  30.0000  151.5500  
4     female  25.0000  151.5500  
...      ...      ...       ...  
1304  female  14.5000   14.4542  
1305  female      NaN   14.4542  
1306    male  26.5000    7.2250  
1307    male  27.0000    7.2250  
1308    male  29.0000    7.8750  

[1309 rows x 6 columns]

6. Удаление
Удалим ненужные столбцы, чтобы нам было проще ориентироваться.

In [56]:
titanic = titanic.replace({"body":"NaN"}, "Not Found")
titanic

pclass  survived                                             name  \
0          1         1                    Allen, Miss. Elisabeth Walton   
1          1         1                   Allison, Master. Hudson Trevor   
2          1         0                     Allison, Miss. Helen Loraine   
3          1         0             Allison, Mr. Hudson Joshua Creighton   
4          1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1304       3         0                             Zabour, Miss. Hileni   
1305       3         0                            Zabour, Miss. Thamine   
1306       3         0                        Zakarian, Mr. Mapriededer   
1307       3         0                              Zakarian, Mr. Ortin   
1308       3         0                               Zimmerman, Mr. Leo   

         sex      age  sibsp  parch  ticket      fare    cabin embarked boat  \
0     female  29.0000      0      0   24160  211.3375       B5        S    2   
1       male   0.9167      1      2  113781  151.5500  C22 C26        S   11   
2     female   2.0000      1      2  113781  151.5500  C22 C26        S  NaN   
3       male  30.0000      1      2  113781  151.5500  C22 C26        S  NaN   
4     female  25.0000      1      2  113781  151.5500  C22 C26        S  NaN   
...      ...      ...    ...    ...     ...       ...      ...      ...  ...   
1304  female  14.5000      1      0    2665   14.4542      NaN        C  NaN   
1305  female      NaN      1      0    2665   14.4542      NaN        C  NaN   
1306    male  26.5000      0      0    2656    7.2250      NaN        C  NaN   
1307    male  27.0000      0      0    2670    7.2250      NaN        C  NaN   
1308    male  29.0000      0      0  315082    7.8750      NaN        S  NaN   

       body                        home.dest  
0       NaN                     St Louis, MO  
1       NaN  Montreal, PQ / Chesterville, ON  
2       NaN  Montreal, PQ / Chesterville, ON  
3     135.0  Montreal, PQ / Chesterville, ON  
4       NaN  Montreal, PQ / Chesterville, ON  
...     ...                              ...  
1304  328.0                              NaN  
1305    NaN                              NaN  
1306  304.0                              NaN  
1307    NaN                              NaN  
1308    NaN                              NaN  

[1309 rows x 14 columns]

7. Замена
Заменим значения NaN в колонке тел на Not Found.

In [57]:
titanic.drop_duplicates()

pclass  survived                                             name  \
0          1         1                    Allen, Miss. Elisabeth Walton   
1          1         1                   Allison, Master. Hudson Trevor   
2          1         0                     Allison, Miss. Helen Loraine   
3          1         0             Allison, Mr. Hudson Joshua Creighton   
4          1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)   
...      ...       ...                                              ...   
1304       3         0                             Zabour, Miss. Hileni   
1305       3         0                            Zabour, Miss. Thamine   
1306       3         0                        Zakarian, Mr. Mapriededer   
1307       3         0                              Zakarian, Mr. Ortin   
1308       3         0                               Zimmerman, Mr. Leo   

         sex      age  sibsp  parch  ticket      fare    cabin embarked boat  \
0     female  29.0000      0      0   24160  211.3375       B5        S    2   
1       male   0.9167      1      2  113781  151.5500  C22 C26        S   11   
2     female   2.0000      1      2  113781  151.5500  C22 C26        S  NaN   
3       male  30.0000      1      2  113781  151.5500  C22 C26        S  NaN   
4     female  25.0000      1      2  113781  151.5500  C22 C26        S  NaN   
...      ...      ...    ...    ...     ...       ...      ...      ...  ...   
1304  female  14.5000      1      0    2665   14.4542      NaN        C  NaN   
1305  female      NaN      1      0    2665   14.4542      NaN        C  NaN   
1306    male  26.5000      0      0    2656    7.2250      NaN        C  NaN   
1307    male  27.0000      0      0    2670    7.2250      NaN        C  NaN   
1308    male  29.0000      0      0  315082    7.8750      NaN        S  NaN   

       body                        home.dest  
0       NaN                     St Louis, MO  
1       NaN  Montreal, PQ / Chesterville, ON  
2       NaN  Montreal, PQ / Chesterville, ON  
3     135.0  Montreal, PQ / Chesterville, ON  
4       NaN  Montreal, PQ / Chesterville, ON  
...     ...                              ...  
1304  328.0                              NaN  
1305    NaN                              NaN  
1306  304.0                              NaN  
1307    NaN                              NaN  
1308    NaN                              NaN  

[1309 rows x 14 columns]

8. Удаление дубликатов.
Удалим дубликаты из датасета, если они есть. По количеству строк понимаем, что дубликатов не было.

In [58]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   pclass     1309 non-null   int64  
 1   survived   1309 non-null   int64  
 2   name       1309 non-null   object 
 3   sex        1309 non-null   object 
 4   age        1046 non-null   float64
 5   sibsp      1309 non-null   int64  
 6   parch      1309 non-null   int64  
 7   ticket     1309 non-null   object 
 8   fare       1308 non-null   float64
 9   cabin      295 non-null    object 
 10  embarked   1307 non-null   object 
 11  boat       486 non-null    object 
 12  body       121 non-null    float64
 13  home.dest  745 non-null    object 
dtypes: float64(3), int64(4), object(7)
memory usage: 143.3+ KB


9. Анализ с помощью info.
После info анализа можем сделать вывод, что из 1309 пассажиров датасета, найденных тел 121, 486 пассажиров спаслось на лодках.

In [59]:
titanic.describe()

pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881135     0.498854     0.385027   
std       0.837836     0.486055    14.413500     1.041658     0.865560   
min       1.000000     0.000000     0.166700     0.000000     0.000000   
25%       2.000000     0.000000    21.000000     0.000000     0.000000   
50%       3.000000     0.000000    28.000000     0.000000     0.000000   
75%       3.000000     1.000000    39.000000     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%       7.895800   72.000000  
50%      14.454200  155.000000  
75%      31.275000  256.000000  
max     512.329200  328.000000

10. Анализ с помощью describe.
После describe анализа видим, что максимальное количество сопровождающих - 9, максимальное количество братьев/сестёр - 8, минимальный возрасть пассажира - меньше месяца, а максимальный - 80 лет. Минимальная цена билета - бесплатно, а максимальная 512 фунтов 

In [60]:
titanic.loc[[34,65,45], ['survived', 'age', 'sex', 'fare']]

survived   age     sex   fare
34         0  42.0    male  26.55
65         1  33.0  female  53.10
45         0  45.0    male  26.55

11. Выборка с помощью loc.
Выберем 3 рандомных пассажира и просмотрим их пол, возраст, цену билета и показатель выживаемости. Делаем вывод, что выжила молодая женщина, с более дорогим билетом, чем остальные 2 человека.

In [61]:
titanic.to_csv("titanic_new.csv", index=False)

12. Сохранение.
Сохраним обновлённый датасет в новый csv файл.


Вывод:
был реализован анализ датасета с пассажирами Титаника. В ходе исследование оказалось, что выживали в основном пассажиры 1 класса, те, у кого дороже стоил билет, женщины и дети. Из представленных в датасете 1309 пассажиров, только около 500 спаслись на лодках. Подавляющее большиство тел не было найдено. Изучены функции describe, info, replace, drop, drop_duplicates, loc, to_csv, sort_values, unique, shape, columns. 